### PR TITLE
sklearnex mention in documentation

### PR DESCRIPTION
**Pull request by Alexsandruss**
Mention of possibility to improve auto-sklearn with [Intel(R) Extension for Scikit-learn*](https://github.com/intel/scikit-learn-intelex) in documentation.
Partially solves https://github.com/automl/auto-sklearn/issues/1184 and https://github.com/automl/auto-sklearn/issues/1171.

### CODE REVIEW

**Comment by eddiebergman** [comment]
Hi @Alexsandruss,

We will discuss this during our meeting next Monday.

Best,
Eddie

**Comment by eddiebergman** [comment]
Hi @Alexsandruss,

I've made a PR #1385 where sklearnex is injected into all of our tests. If this passes without issues then we will pull in the documentation. However we would change it to be in the manual and remove it from the `README.md`.

For the sake of brevity, I'll likely just do this in a separate PR and manually copy your text over. I'll be sure to include your contributions in the Release notes.

I do however have some questions which @mfeurer might be interested in too. We use `dask` for distributing workloads which by default creates seperate processes or allows users to use compute clusters. I feel like `sklearnex` does not handle this, especially consdering this piece of [documentation I found](https://intel.github.io/scikit-learn-intelex/distributed-mode.html?highlight=distribute). Can you confirm this in any manner?

If it doesn't that is still fine but we would have to explicitly advertise this fact when mentioning `sklearnex`, not to confuse users who might want a highly distributed workflow and `sklearnex`.

**Comment by Alexsandruss** [comment]
Hi @eddiebergman,

Can you share the reason why short sklearnex mention with link to manual can't be included in README.md or start section of docs?

Currently, sklearnex doesn't support distributed mode, but it is in development plans.

**Comment by eddiebergman** [comment]
Hi @Alexsandruss,

While it could be beneficial for performance, the `README.md` is short and concise, purely about `autosklearn`. If we include `sklearnex`, we could then also advertise `SHAP` with `autosklearn` because interoperability is important or other optimizations that can be done through `SMAC`. There are quite a few possibilities so we will keep this to the manual. If we are to ever integrate it ([as autogluon does](https://github.com/awslabs/autogluon/blob/master/tabular/setup.py#L45)) then we would bump it up to the installation steps.

We have tested it and it seems to work (without issues) so we will include some documentation and an example. For this example, do you recommend `patch_sklearn` or `patch_global_sklearn`, would this help with a distributed setting?

Further, we are currently benchmarking it. If this provides a significant performance improvement, we will discuss including it as `autogluon` does above and as mentioned, include a point in the `install.rst` and `manual.rst`.

**Comment by mfeurer** [comment]
I'm worried that there's some misunderstanding about `dask` here. We use `dask` to distribute the workload across different workers. These workers can be on the same machine, and also different machines. We do not use dask to distribute the workload of individual models, as for example [XGBoost](https://xgboost.readthedocs.io/en/stable/tutorials/dask.html) does. However, a common feature is that all workers will spawn subprocesses using the `fork_server`. The exact questions would be "does sklearnex work in such a subprocess on the same machine" and "does sklearnex work in such a subprocess on a different machine".

**Comment by Alexsandruss** [comment]
`patch_sklearn` is recommended and not connected to distributed mode setup.

sklearnex doesn't directly support dask but works in subprocess. For example, this code will be accelerated:

```python
from sklearnex import patch_sklearn
patch_sklearn()
from sklearn.linear_model import Ridge

...

ridge = Ridge()
grid_search = GridSearchCV(ridge, parameters, n_jobs=-1, verbose=1, cv=3, refit=False)

with joblib.parallel_backend('dask'):
    grid_search.fit(x, y)
```

This was tested on one machine, but not on different machines.

**Comment by eddiebergman** [comment]
That's good to know, thanks very much for that :) We'll get back to you on Monday with more!

**Comment by Alexsandruss** [comment]
@eddiebergman, any updates from Monday meeting?

**Comment by eddiebergman** [comment]
Hi @Alexsandruss,

I was going to post earlier but I wanted to wait until next Monday. Good news is that it seems to work fine given our default distributed method with dask. Just using the `n_jobs` parameter without a custom dask client seemed to have sklearnex work.

Unexpectedly, we got some slight numerical difference with our unit tests using it, meaning it's unlikely we'll bring it in directly anytime soon but the differences were marginal mostly and didn't seem to indicate much.

I did however perform a benchmark on [`automl_benchmark`](https://openml.github.io/automlbenchmark/automl_overview.html), particularly [`openml/s/218`](https://www.openml.org/s/218). For some reason, with sklearnex enabled, no models would be found for dataset [189355](https://www.openml.org/t/189355). I didn't have the tools to investigate this fully but would you have any intution for why this might be. It's a particularly large dataset with respect to the rest of `openml/s/218` but I'm not sure why this would cause an issue. We are going to discuss this next Monday and I will post a log here of the output once we have. I couldn't find anything that would indicate this failure but perhaps you would have better luck. I also ran a stable version without sklearnex at the same version and there were no failures.

I will rebase the sklearnex version to our latest and run the benchmark again on Monday.

Best,
Eddie

**Comment by eddiebergman** [comment]
Update:

I've rebased sklearnex to be on our latest branch and will run the benchmark again.
In the meantime, there also seems to be failing unit tests with respect to Python 3.10. I've raise an issue [here.](https://github.com/intel/scikit-learn-intelex/issues/968)

**Comment by eddiebergman** [comment]
Hello @Alexsandruss,

I've reran the benchmark and it still causes occasional failures on the dataset mentioned above. The issue that arises is that autosklearn tends to not find a model for some reason in some off cases.

The error:
```
[WARNING] [2022-02-16 13:17:48,421:Client-EnsembleBuilder] No models better than random - using Dummy loss!Number   of models besides current dummy model: 1. Number of dummy models: 1 
```

Other folds of the dataset have a similar error in which it indicates no models are found.

I've attached the full log for that fold with anonymized paths
[tmp.log](https://github.com/automl/auto-sklearn/files/8110943/tmp.log)

We've decided we will include documentation refering to sklearnex, indicating that it can improve performance as your previous original PR indicated and that it seems to work in most settings. However we can't make any guarantees on it's stability. It will be mentioned in the [other](https://automl.github.io/auto-sklearn/master/manual.html#other) section of the manual, give it an FAQ section in [other](https://automl.github.io/auto-sklearn/master/faq.html#other) and include an example for how to use it.

Best,
Eddie

**Comment by jmhatre-work** [comment]
Hi @eddiebergman,

 We tried running automlbenchmark and found fallbacks due to reasons that we do not support conditions in our algorithms as mentioned [here](https://github.com/automl/auto-sklearn/issues/1171#issuecomment-882562616). 
 [Here ](https://intel.github.io/scikit-learn-intelex/algorithms.html) is list of supported features for algorithms in sklearnex. 
 
 We ran [tests ](https://colab.research.google.com/drive/1yQw47QtMbqiMi-iD_-nBNPWCahUaNm_W#scrollTo=Hdl2KgsiO-9l&uniqifier=2) for algorithms and did not find (Client-EnsembleBuilder] No models better than random - using Dummy loss!Number   of models besides current dummy model: 1. Number of dummy models: 1 ) 

Furthermore, I was wondering if there any more issues to address for sklearnex to be mentioned in  [other ](https://automl.github.io/auto-sklearn/master/manual.html#other) as discussed in previous comment.
